LongWarOfTheChosen/Src/LW_Overhaul/Classes/UIPersonnel_SoldierListItem_LW.uc

```diff
@@ -386,13 +398,67 @@ function AddClassColumnIcons(XComGameState_Unit Unit)
 	WillValue.SetHtmlText(class'UIUtilities_Text'.static.GetColoredText(GetWillString(Unit), Unit.GetMentalStateUIState()));
 }
 
+simulated function int GetWillValueColour(XComGameState_Unit Soldier, bool Disabled, bool Focused)
+{
+	// KDM : Colour the will value text according to these, ordered, rules :
+	// 1.] If the soldier row is disabled, use the disabled, greyish, text colour.
+	// 2.] If the soldier row is focused, use black.
+	// 3.] Use green/yellow/red depending upon the mental status of the soldier associated with this row.
+	if (Disabled)
+	{
+		return eUIState_Disabled;
+	}
+	else if (Focused)
+	{
+		return -1;
+	}
+	else
+	{
+		return int(Soldier.GetMentalStateUIState());
+	}
+}
+
+simulated function string GetCombatIntelligenceColour(XComGameState_Unit Soldier, bool Disabled, bool Focused)
+{
+	// KDM : Colour the combat intelligence image according to these, ordered, rules :
+	// 1.] If the soldier row is disabled, use the disabled, greyish, text colour : 0x828282.
+	// 2.] If the soldier row is focused, use black : 0x000000.
+	// 3.] Use red/orange/yellow/green blue depending upon the soldier's combat intelligence level.
+	if (Disabled)
+	{
+		return class'UIUtilities_Colors'.const.DISABLED_HTML_COLOR;
+	}
+	else if (Focused)
+	{
+		return class'UIUtilities_Colors'.const.BLACK_HTML_COLOR;
+	}
+	else
+	{
+		switch (Soldier.ComInt)
+		{
+		case eComInt_Standard:
+			return CI_STANDARD_COLOUR;
+		case eComInt_AboveAverage:
+			return CI_ABOVEAVERAGE_COLOUR;
+		case eComInt_Gifted:
+			return CI_GIFTED_COLOUR;
+		case eComInt_Genius:
+			return CI_GENIUS_COLOUR;
+		case eComInt_Savant:
+			return CI_SAVANT_COLOUR;
+		default:
+			return class'UIUtilities_Colors'.const.BLACK_HTML_COLOR;
+		}
+	}
+}
+
 simulated function UpdateItemsForFocus(bool Focussed)
 {
-	local int iUIState;
-	local XComGameState_Unit Unit;
 	local bool bReverse;
+	local int iUIState;
 	local string Aim, Defense, Health, Mobility, Will, Hack, Dodge, Psi, ComIntColor;
```

Comment:
`ComIntColor` is no longer used in this function it seems and is causing a compiler warning.